graphql_compiler/interpreter/__init__.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020-present Kensho Technologies, LLC.
+"""Tools for constructing high-performance query interpreters over arbitrary schemas.
+
+While GraphQL compiler's database querying capabilities are sufficient for many use cases, there are
+many types of data querying for which the compilation-based approach is unsuitable. A few examples:
+- data accessible via a simple API instead of a rich query language,
+- data represented as a set of files and directories on a local disk,
+- data produced on-demand by running a machine learning model over some inputs.
+
+The data in each of these cases can be described by a valid schema, and users could write
+well-defined and legal queries against that schema. However, the execution of such queries cannot
+proceed by compiling them to another query language -- no such target query language exists.
+Instead, the queries need to be executed using an *interpreter*: a sophisticated piece of code
+that executes queries incrementally in a series of steps, such as "fetch the value of this field"
+or "filter out this data point if its value is less than 5."
+
+Some parts of the interpreter (e.g. "fetch the value of this field") obviously need to be aware of
+the schema and the underlying data source. Other parts (e.g. "filter out this data point") are
+schema-agnostic -- they work in the same way regardless of the schema and data source. This library
+provides efficient implementations of all schema-agnostic interpreter components. All schema-aware
+logic is abstracted into the straightforward API of the InterpreterAdapter class, which should be
```

Comment:
```suggestion
logic is abstracted into the straightforward, four-method API of the InterpreterAdapter class, which should be
```
Although you mention the compactness & simplicity of the API footprint below, I think it's worth mentioning here, too, as an enticement to the prospective user to keep reading.  (It is EXTREMELY enticing!)